pulsar-client-api/src/main/java/org/apache/pulsar/client/api/schema/RecordSchemaBuilder.java

```diff
@@ -54,7 +54,17 @@
      * @param genericSchema schema of the field
      * @return field schema builder to build the field.
      */
-    FieldSchemaBuilder field(String fieldName, GenericSchema genericSchema);
+    FieldSchemaBuilder field(String fieldName, org.apache.pulsar.client.api.Schema<?> genericSchema);
+
+    /**
+     * Add a field with the given name and genericSchema to the record.
+     *
+     * @param fieldName name of the field
+     * @param genericSchema schema of the field
+     * @return field schema builder to build the field.
+     */
+    @Deprecated
+    FieldSchemaBuilder field(String fieldName, GenericSchema<?> genericSchema);
```

Comment:
Any reason why do you deprecated it?

Reply:
I have deprecated it in favour of the new method that allows a `Schema<?>`
the new method is more general, and it should be preferred.
I cannot remove the one with `GenericSchema<?>` in order to keep binary compatibility with compiled code